flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -413,6 +406,45 @@ private static void generateRandomDirs(File dir, int numFiles, int numDirs, int
 		return jarFiles;
 	}
 
+	/**
+	 * Generate some directories in a original directory based on the {@code baseDir}.
+	 * @param baseDir the directory where the original directory is generated
+	 * @return The relative path of the original directory, which is based on the {@code baseDir}.
+	 * @throws IOException if I/O error occurs while generating the directories
+	 */
+	public static java.nio.file.Path prepareTestDirectories(final java.nio.file.Path baseDir) throws IOException {
```

Comment:
No need to make it `public` and I'd probably also drop the `static`.
I'd also make the method name more explicit like `prepareTestDirectoriesForCompression` because there are multiple other test cases in this class.